### PR TITLE
XEP-0198: Add section defining SASL2 and BIND2 interaction

### PR DESCRIPTION
**Pull request by tmolitor-stud-tu**
The title says it all: I've added a section specifying interaction with the upcoming versions of the SASL2 (XEP-0388) and BIND2 (XEP-0386) XEPs.

See #1214 for details about the SASL2 changes.

### CODE REVIEW

**Comment by horazont** [comment]
Even though this is pending the acceptance of #1214, I'm still submitting this to council to judge the general approach of having this in '198 as well as any concerns which may exist under the precondition of #1214 being merged.

**Comment by tmolitor-stud-tu** [comment]
For reference, here is the rendered version: https://dyn.eightysoft.de/final/xep-0198.html#inline

File: xep-0198.xml [hunk 1]
```diff
@@ -552,8 +559,95 @@
   </section2>
 </section1>
 
+<section1 topic="SASL2 And BIND2 Interaction" anchor="inline">
+  <p class='box'>This section is about &xep0388; (SASL2) and &xep0386; (BIND2) interaction. You don't have to implement this if you don't implement SASL2 and BIND2.</p>
+  <p>&xep0388; (SASL2) describes a way of inlining the stream resumption into the authentication process to reduce the round-trips needed for authentication and subsequent stream resumption. Similarly &xep0386; (BIND2) defines a way of inlining the stream management &lt;enable/> into the resource binding process to reduce round-trips.</p>
+  <section2 topic="Enable Stream Management Inline" anchor="inline-enable">
+    <p>To indicate support for inlining the activation of Stream Management into the resource binding process, the server adds a &lt;feature/> element in the namespace "urn:xmpp:sm:3" into the &lt;inline/> element of BIND2 which is sent in the stream features.</p>
```
Comment (mar-v-in) [inline xep-0198.xml:566]: should be `server adds a &lt;feature/> element with var attribute set to "urn:xmpp:sm:3" in the &lt;inline/> element`

**Review by mar-v-in — CHANGES_REQUESTED** [review]

**Comment by iNPUTmice** [comment]
Hi @tmolitor-stud-tu the feedback @mar-v-in provided hasn’t been addressed yet. This is why council has rejected this for now. But once this minor thing is fixed I can put it back on the agenda.

**Comment by tmolitor-stud-tu** [comment]
@iNPUTmice I'm sorry, I somehow missed that feedback. Should be addressed now :)

File: xep-0198.xml [hunk 2]
```diff
@@ -552,8 +559,95 @@
...
+    <p>To indicate support for inlining the activation of Stream Management into the resource binding process, the server adds a &lt;feature/> element with var attribute set to "urn:xmpp:sm:3" in the &lt;inline/> element of BIND2 which is sent in the stream features.</p>
+    <p>If the client wishes to begin a new session (i.e. it has no prior session to resume), it simply includes the &lt;enable/> element defined by this specification in its &xep0386; &lt;bind/> request which itself is a child of the &lt;authenticate/> element of SASL2.</p>
+    <section3 topic="Failed To Enable Stream Management">
+      <p>In the unexpected case where the server was able to bind a resource for the client, but unable to enable stream management, it will include a &lt;failed/> element as defined by this specification within the &lt;bound/> response defined by &xep0386;.</p>
+    </section3>
+  </section2>
+  <section2 topic="Inline Stream Resumption" anchor="inline-resume">
+    <p>To indicate support for inlining stream resumption into the authentication process, the server adds a &lt;resume/> element in the namespace "urn:xmpp:sm:3" to the &lt;inline/> element of SASL2.</p>
```
Comment (iNPUTmice) [inline xep-0198.xml:573]: The example uses a `<sm/>` element and so do Conversations and prosody

**Comment by iNPUTmice** [comment]
Merged as #1294

**Pull request closed without merging**